node/node.go

```diff
@@ -129,7 +130,8 @@ type Node struct {
 	err              error
 	agent            *agent.Agent
 	manager          *manager.Manager
-	notifyNodeChange chan *api.Node // used to send role updates from the dispatcher api on promotion/demotion
+	notifyNodeChange chan *agent.NodeChanges // used to send role updates from the dispatcher api on promotion/demotion
+	notifyTLSChange  chan *api.NodeTLSInfo   // used to send tls info to the agent on TLS credential when the node changes its trust root
```

Comment:
Sorry the node/security config parts of the PR aren't quite done - I was going to set this to be the watch channel instead.